rules/link-event.js

```diff
@@ -0,0 +1,100 @@
+const {
+  groupBy
+} = require('min-dash');
+
+const {
+  is
+} = require('bpmnlint-utils');
+
+
+/**
+ * A rule that verifies that link events are properly used.
+ *
+ * This implies:
+ *
+ *   * for every link throw there exists a link catch within
+ *     the same scope, and vice versa
+ *   * there exists only a single pair of [ throw, catch ] links
+ *     with a given name, per scope
+ *   * link events have a name
+ *
+ */
+module.exports = function() {
+
+  function check(node, reporter) {
+
+    if (!is(node, 'bpmn:FlowElementsContainer')) {
+      return;
+    }
+
+    const links = (node.flowElements || []).filter(isLinkEvent);
+
+    for (const link of links) {
+      if (!link.name) {
+        reporter.report(link.id, 'Link event is missing name');
+      }
+    }
+
+    const names = groupBy(links, (link) => link.name);
+
+    for (const [ name, events ] of Object.entries(names)) {
+
+      // ignore unnamed (validated earlier)
+      if (!name) {
+        continue;
+      }
+
+      // missing catch event
+      if (events.length === 1) {
+        const event = events[0];
+
+        reporter.report(event.id, `Link ${isThrowEvent(event) ? 'catch' : 'throw' } event with name <${ name }> missing in scope`);
+      }
+
+      const throwEvents = events.filter(isThrowEvent);
+
+      if (throwEvents.length > 1) {
+        for (const event of throwEvents) {
+          reporter.report(event.id, `Duplicate link ${isThrowEvent(event) ? 'throw' : 'catch' } event with name <${name}> in scope`);
+        }
+      }
```

Comment:
We already validated that this is a throw event earlier, right? So the message name could just be:
```suggestion
      const throwEvents = events.filter(isThrowEvent);

      if (throwEvents.length > 1) {
        for (const event of throwEvents) {
          reporter.report(event.id, 'Duplicate link throw event with name <${name}> in scope');
        }
      }
```

Same for catch events below.

Reply:
Fixed.